src/test/groovy/platform/edxPlatformTestSubsetSpec.groovy

```diff
@@ -0,0 +1,226 @@
+package platform
+
+import groovy.util.slurpersupport.GPathResult
+import groovy.util.slurpersupport.Node
+import javaposse.jobdsl.plugin.JenkinsJobManagement
+import javaposse.jobdsl.dsl.GeneratedItems
+import javaposse.jobdsl.dsl.GeneratedJob
+import javaposse.jobdsl.dsl.DslScriptLoader
+import javaposse.jobdsl.dsl.JobManagement
+import org.junit.ClassRule
+import org.jvnet.hudson.test.JenkinsRule
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class edxPlatformTestSubsetSpec extends Specification {
+
+    @Shared
+    @ClassRule
+    JenkinsRule jenkinsRule = new JenkinsRule()
+
+    def @Shared DslScriptLoader loader
+    def @Shared JobManagement jm
+
+    // The DSL script under test
+    def @Shared File dslScript = new File('platform/jobs/edxPlatformTestSubset.groovy')
+    def @Shared String baseJobName = "edx-platform-test-subset"
+    def @Shared String baseSecretPath = "src/test/resources/platform/secrets"
+    // Secret variable used in DSL script
+    def @Shared String secretVar = "EDX_PLATFORM_TEST_SUBSET_SECRET"
+    
+    def @Shared List pList = [ 'com.cloudbees.plugins.credentials.CredentialsProvider.Delete:edx',  
+                                'com.cloudbees.plugins.credentials.CredentialsProvider.ManageDomains:edx', 'hudson.model.Item.Read:edx', 
+                                'hudson.model.Item.Configure:edx', 'hudson.model.Item.Workspace:edx', 'hudson.model.Run.Delete:edx', 
+                                'hudson.model.Item.Discover:edx', 'com.cloudbees.plugins.credentials.CredentialsProvider.View:edx', 
+                                'hudson.model.Item.Build:edx', 'com.cloudbees.plugins.credentials.CredentialsProvider.Create:edx', 
+                                'hudson.model.Item.Cancel:edx', 'hudson.model.Item.Delete:edx', 'hudson.model.Run.Update:edx', 
+                                'com.cloudbees.plugins.credentials.CredentialsProvider.Update:edx' ]
+
+    /*
+    * Helper function: loadSecret
+    * return a JenkinsJobManagement object containing a mapping of secret variables to secret values
+    */
+    JenkinsJobManagement loadSecret(secretKey, secretValue) {
+        Map<String,String> envVars = new HashMap<String,String>()
+        envVars.put(secretKey, secretValue)
+        JenkinsJobManagement jjm = new JenkinsJobManagement(System.out, envVars, new File('.'))
+        return jjm
+    }
+
+
+    /**
+    * Using an evironment variable that points to a non-existent secret file, ensure that no
+    * jobs are created.
+    **/
+    void 'test non-existent secret file is handled correctly'() {
+
+        setup:
+        String secretPath = baseSecretPath + "/non-existent-file.yml"
+        jm =  loadSecret(secretVar, secretPath)
+        loader = new DslScriptLoader(jm)
+
+        when:
+        GeneratedItems generatedItems = loader.runScript(dslScript.text)
+
+        then:
+        generatedItems.jobs.size() == 0
```

Comment:
Technically, this would throw an error (File Not Found or IOException). However, we have encapsulated the Yaml secret parsing within a try block that returns a "1" if any exception is caught. The reason for this is that returning a 1 (or any int besides 0) will fail the Jenkins job, and this is desirable, so that the DSL engine doesnt continue running other scripts. Unfortunately, because we return a 1, the call to loader.runScript() doesn't funnel that back to the test. I don't like it, but until I can come up with a better way, it "seems" the only method to test for this type of failure is by verifying nothing was successful.